commit 2a1b95dd97c6435c599f2e2f87c39cbb7511e682
Author: Pavel Shilovsky <pshilov@microsoft.com>
Date:   Thu Jan 19 13:53:15 2017 -0800

    CIFS: Fix splice read for non-cached files
    
    commit 9c25702cee1405099f982894c865c163de7909a8 upstream.
    
    Currently we call copy_page_to_iter() for uncached reading into a pipe.
    This is wrong because it treats pages as VFS cache pages and copies references
    rather than actual data. When we are trying to read from the pipe we end up
    calling page_cache_pipe_buf_confirm() which returns -ENODATA. This error
    is translated into 0 which is returned to a user.
    
    This issue is reproduced by running xfs-tests suite (generic test #249)
    against mount points with "cache=none". Fix it by mapping pages manually
    and calling copy_to_iter() that copies data into the pipe.
    
    Signed-off-by: Pavel Shilovsky <pshilov@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1249_2a1b95dd97c6435c599f2e2f87c39cbb7511e682.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1249_2a1b95dd97c6435c599f2e2f87c39cbb7511e682.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
